core/src/main/java/org/apache/carbondata/core/datastore/blocklet/EncodedBlocklet.java

```diff
@@ -87,19 +91,24 @@ private void addPageMetadata(EncodedTablePage encodedTablePage) {
    * @param encodedTablePage
    * encoded table page
    */
-  private void addEncodedMeasurePage(EncodedTablePage encodedTablePage) {
+  private void addEncodedMeasurePage(EncodedTablePage encodedTablePage,
+      Map<String, LocalDictionaryGenerator> localDictionaryGeneratorMap) {
     // for first page create new list
     if (null == encodedMeasureColumnPages) {
       encodedMeasureColumnPages = new ArrayList<>();
       // adding measure pages
       for (int i = 0; i < encodedTablePage.getNumMeasures(); i++) {
-        BlockletEncodedColumnPage blockletEncodedColumnPage = new BlockletEncodedColumnPage(null);
-        blockletEncodedColumnPage.addEncodedColumnColumnPage(encodedTablePage.getMeasure(i));
+        BlockletEncodedColumnPage blockletEncodedColumnPage = new BlockletEncodedColumnPage(null,
+            Boolean.parseBoolean(CarbonProperties.getInstance()
```

Comment:
What if the configuration is changed during data loading? So that each column page will have different configuration, will this be OK?

Reply:
i will take the performnce report with this change, then we can decide

Reply:
@xuchuanyin i have tested and results i have published, i think we can keep it as property and make default by true, as we getting good result with respect to memory